### PR TITLE
Minor qmflows.cp2k_utils documentation cleaning

### PR DESCRIPTION
**Pull request by BvB93**
* Replaced most sphinx :func:`func<module.func>` statements with the shorter and more readable :func:`~module.func` syntax.
* Added the, previously missing, ``prm_to_df()`` function to the module-level docstring.
* Replaced all bare ``# noqa`` comments with error-code-specific ones (*e.g.* ``# noqa: E501``).
* Added a missing f-string in the ``_cp2k_keys_alias()`` function.
* Minor typing adjustments.

### CODE REVIEW

File: src/qmflows/cp2k_utils.py [hunk 1]
```diff
@@ -243,16 +248,16 @@ def set_prm(settings: Settings, key: Union[str, Tuple[str, ...]],
     :data:`CP2K_KEYS_ALIAS` : :class:`dict` [:class:`str`, :class:`tuple` [:class:`str`, ...]]
         A dictionary mapping ``key_path`` aliases to the actual keys.
 
-    """  # noqa
+    """  # noqa: E501
     if isinstance(value, abc.Sequence):
-        for prm_map in value:
+        for prm_map in value:  # type: Union[MappingScalar, MappingSequence]
```
Comment (BvB93) [inline src/qmflows/cp2k_utils.py:253]: A type comment. In this example it is used for assigning a type to the elements of a sequences (*i.e.* ``prm_map``).

File: src/qmflows/cp2k_utils.py [hunk 2]
```diff
@@ -243,16 +248,16 @@ def set_prm(settings: Settings, key: Union[str, Tuple[str, ...]],
...
             set_prm(settings, key, prm_map, mol)
         return
     else:
         prm_map = copy.copy(value)
 
     try:
-        prm_key = prm_map.pop('param')
+        prm_key = cast(Union[str, Sequence[str]], prm_map.pop('param'))
```
Comment (BvB93) [inline src/qmflows/cp2k_utils.py:260]: The [``cast()``](https://docs.python.org/3/library/typing.html#typing.cast) function is used for forcefully assign a certain type to a variable; it does nothing at runtime.

**Review by felipeZ — COMMENTED** [review]
Indeed ~ makes the documentation links easier to read

File: src/qmflows/cp2k_utils.py (hunk 1 above)
Comment (felipeZ) [reply src/qmflows/cp2k_utils.py:253]: I think it is good for documentation purposes

File: src/qmflows/cp2k_utils.py (hunk 2 above)
Comment (felipeZ) [reply src/qmflows/cp2k_utils.py:260]: Good to know, because in C++ for example it cast one type to another

**Pull request merged**